js/bootstrap-datepicker.js

```diff
@@ -842,14 +842,17 @@
 		},
 
 		fillMonths: function(){
-      var localDate = this._utc_to_local(this.viewDate);
-			var html = '';
-			var focused;
+			var focused, node, localDate = this._utc_to_local(this.viewDate);
+			this.picker.find('.datepicker-months td').html('');
 			for (var i = 0; i < 12; i++){
 				focused = localDate && localDate.getMonth() === i ? ' focused' : '';
-				html += '<span class="month' + focused + '">' + dates[this.o.language].monthsShort[i] + '</span>';
+				node = $('<span>');
+				node.data('date', {day: 1, month: i, year: this.viewDate.getUTCFullYear()});
+				node.attr('class', 'month' + focused);
+				node.html(dates[this.o.language].monthsShort[i]);
+				node.on('click', $.proxy(this.cellClick, this));
```

Comment:
Yeah I though about it the other day and it hit me what @vsn4ik was talking about.  
The issue is that by design it is not easy to catch the event when the DOM is customized.  
Comprehensive drawing:  
![image](https://cloud.githubusercontent.com/assets/1201743/23754237/94454bf2-04dc-11e7-99f4-a9f84d1e08b3.png)

Before it was easy, you can bind on the container, and get the source of the event with `currentTarget`.  
Now `currentTarget` will return the custom DOM nodes instead of the td cell.

The easiest and safest way is then to bind the event to the cell, so `target` will equals the cell, always.    
If we keep the event bound to the container, it mean we'll have to lookup for the cell between the  `target` and the `currentTarget`.  

So the question is, which is the best performance wise between binding 90 events or recursively find  a DOM element.